hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -2191,6 +2208,44 @@ public void shutdown() {
     tracer.close();
   }
 
+  public void startCheckDiskThread() {
```

Comment:
Suggest to use `ScheduledExecutorService` to schedule this task at fixed rate. since,
1. same target,
2. simpler,

`checkDiskMutex`, it seems to me that it it no need to concurrency control.